<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="A.DeMartin" />

<meta name="date" content="2025-05-16" />

<title>E18_mLN_iLN</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">LNmLToRev</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">E18_mLN_iLN</h1>
<h4 class="author">A.DeMartin</h4>
<h4 class="date">2025-05-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-20
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>LNmLToRev/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/LNmLToRev/data/LNmLToRev_allmerged_seurat.rds
</td>
<td style="text-align:left;">
data/LNmLToRev_allmerged_seurat.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/LNmLToRev/data/LNmLToRev_E18fil_seurat.rds
</td>
<td style="text-align:left;">
data/LNmLToRev_E18fil_seurat.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/LNmLToRev/data/LNmLToRev_E18fil_integrated_seurat.rds
</td>
<td style="text-align:left;">
data/LNmLToRev_E18fil_integrated_seurat.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/LNmLToRev/data/markerGenes_E18fil_integrated_intCluster
</td>
<td style="text-align:left;">
data/markerGenes_E18fil_integrated_intCluster
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomangeldemartinLNmLToRevtreef0ae62604be035da45cfd21b95b314fdfcbe6b37targetblankf0ae626a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/angeldemartin/LNmLToRev/tree/f0ae62604be035da45cfd21b95b314fdfcbe6b37" target="_blank">f0ae626</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomangeldemartinLNmLToRevtreef0ae62604be035da45cfd21b95b314fdfcbe6b37targetblankf0ae626a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/angeldemartin/LNmLToRev/tree/f0ae62604be035da45cfd21b95b314fdfcbe6b37" target="_blank">f0ae626</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    data/files/
    Ignored:    data/tradeSEQ/

Unstaged changes:
    Modified:   analysis/E18_mLN_iLN.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/E18_mLN_iLN.Rmd</code>) and HTML
(<code>docs/E18_mLN_iLN.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/analysis/E18_mLN_iLN.Rmd" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
<td>
may20-4
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/angeldemartin/LNmLToRev/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/E18_mLN_iLN.html" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
<td>
may20-4
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/angeldemartin/LNmLToRev/7eb18ba49db6016e50aa819397bd8c950ca38572/docs/E18_mLN_iLN.html" target="_blank">7eb18ba</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
<td>
may20-1
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/79d33a83516fccdb2173a5f4acb87d7f13c1e18e/analysis/E18_mLN_iLN.Rmd" target="_blank">79d33a8</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
<td>
may20
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/d715ca309a2118f8f5cb9ae8d3f89a95b21a7054/analysis/E18_mLN_iLN.Rmd" target="_blank">d715ca3</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-16
</td>
<td>
may16
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-and-preprocess" class="section level2">
<h2>load and preprocess</h2>
<div id="load-packages" class="section level3">
<h3>load packages</h3>
</div>
<div id="set-color-vectors" class="section level3">
<h3>set color vectors</h3>
<pre class="r"><code>coltimepoint &lt;- c(&quot;#440154FF&quot;, &quot;#3B528BFF&quot;, &quot;#21908CFF&quot;, &quot;#5DC863FF&quot;)
names(coltimepoint) &lt;- c(&quot;E18&quot;, &quot;P7&quot;, &quot;3w&quot;, &quot;8w&quot;)

collocation &lt;- c(&quot;#61baba&quot;, &quot;#ba6161&quot;)
names(collocation) &lt;- c(&quot;iLN&quot;, &quot;mLN&quot;)</code></pre>
</div>
<div id="load-object" class="section level3">
<h3>load object</h3>
</div>
<div id="subset-e18" class="section level3">
<h3>subset E18</h3>
</div>
<div id="plot-umaps" class="section level3">
<h3>plot umaps</h3>
<div id="clustering" class="section level4">
<h4>clustering</h4>
<pre class="r"><code>Idents(seuratA) &lt;- seuratA$RNA_snn_res.0.25
DimPlot(seuratA, reduction = &quot;umap&quot;, group.by = &quot;RNA_snn_res.0.25&quot; ,
        pt.size = 0.1, label = T, shuffle = T) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;umap1&quot;) +
  ylab(&quot;umap2&quot;)</code></pre>
</div>
<div id="location" class="section level4">
<h4>location</h4>
<pre class="r"><code>DimPlot(seuratA, reduction = &quot;umap&quot;, group.by = &quot;location&quot; ,
        pt.size = 0.1, label = T, shuffle = T) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;umap1&quot;) +
  ylab(&quot;umap2&quot;)</code></pre>
</div>
</div>
<div id="calculate-cluster-marker-genes-pre-filter"
class="section level3">
<h3>calculate cluster marker genes pre filter</h3>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratA)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) 

selGenes &lt;- data.frame(geneID=c(&quot;Ptprc&quot;, &quot;Msln&quot;, &quot;Mki67&quot;, &quot;Kcnn3&quot;, &quot;Tcf21&quot;, &quot;Pecam1&quot;, &quot;Lyve1&quot;, &quot;Ccl21a&quot;, &quot;Icam1&quot;, &quot;Cd34&quot;)) %&gt;% 
  left_join(., genes, by = &quot;geneID&quot;) 

pList &lt;- sapply(selGenes$gene, function(x){
  p &lt;- FeaturePlot(seuratA, reduction = &quot;umap&quot;, 
            features = x, 
            cols=c(&quot;lightgrey&quot;,&quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
  plot(p)
})</code></pre>
</div>
</div>
<div id="filter" class="section level2">
<h2>filter</h2>
<div id="filter-object" class="section level3">
<h3>filter object</h3>
<pre class="r"><code>## filter out and Ptprc (cluster #7 and #14)


##mesothelial cells (cluster#4) 
##proliferating cells (cluster #1)
##epithelial/neuronal (cluster #5 and #13)
##pancreatic cells (#) 

table(seuratA$RNA_snn_res.0.25)
seuratF &lt;- subset(seuratA, RNA_snn_res.0.25 %in% c(&quot;7&quot;, &quot;14&quot;), invert = TRUE)
table(seuratF$RNA_snn_res.0.25)

seuratE18fil &lt;- seuratF
remove(seuratF)</code></pre>
</div>
<div id="rerun-after-fil" class="section level3">
<h3>rerun after fil</h3>
</div>
<div id="load-object-fil" class="section level3">
<h3>load object fil</h3>
<pre class="r"><code>fileNam &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/LNmLToRev/data/LNmLToRev_E18fil_seurat.rds&quot;
seuratE18fil &lt;- readRDS(fileNam)</code></pre>
</div>
</div>
<div id="dimplot" class="section level2">
<h2>dimplot</h2>
<div id="clustering-1" class="section level3">
<h3>clustering</h3>
<pre class="r"><code>Idents(seuratE18fil) &lt;- seuratE18fil$RNA_snn_res.0.25
colPal &lt;- c(&quot;#DAF7A6&quot;, &quot;#FFC300&quot;, &quot;#FF5733&quot;, &quot;#C70039&quot;, &quot;#900C3F&quot;, &quot;#b66e8d&quot;,
            &quot;#61a4ba&quot;, &quot;#6178ba&quot;, &quot;#54a87f&quot;,  &quot;#25328a&quot;,
            &quot;#b6856e&quot;, &quot;#0073C2FF&quot;, &quot;#EFC000FF&quot;, &quot;#868686FF&quot;, &quot;#CD534CFF&quot;, 
            &quot;#7AA6DCFF&quot;, &quot;#003C67FF&quot;, &quot;#8F7700FF&quot;, &quot;#3B3B3BFF&quot;, &quot;#A73030FF&quot;,
            &quot;#4A6990FF&quot;)[1:length(unique(seuratE18fil$RNA_snn_res.0.25))]
names(colPal) &lt;- unique(seuratE18fil$RNA_snn_res.0.25)

DimPlot(seuratE18fil, reduction = &quot;umap&quot;, group.by = &quot;RNA_snn_res.0.25&quot;,
        cols = colPal, label = TRUE)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN.Rmd/dimplot%20clustering%20E18fil-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-dimplot-clustering-E18fil-1">
Past versions of "dimplot clustering E18fil-1.png"
</button>
</p>
<div id="fig-no-spaces-dimplot-clustering-E18fil-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/7eb18ba49db6016e50aa819397bd8c950ca38572/docs/figure/E18_mLN_iLN.Rmd/dimplot clustering E18fil-1.png" target="_blank">7eb18ba</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="location-1" class="section level3">
<h3>location</h3>
<pre class="r"><code>DimPlot(seuratE18fil, reduction = &quot;umap&quot;, group.by = &quot;location&quot;,
        cols = collocation)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN.Rmd/dimplot%20loction%20E18fil-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-dimplot-loction-E18fil-1">
Past versions of "dimplot loction E18fil-1.png"
</button>
</p>
<div id="fig-no-spaces-dimplot-loction-E18fil-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/7eb18ba49db6016e50aa819397bd8c950ca38572/docs/figure/E18_mLN_iLN.Rmd/dimplot loction E18fil-1.png" target="_blank">7eb18ba</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-object-fil" class="section level3">
<h3>save object fil</h3>
<pre class="r"><code>saveRDS(seuratE18fil, file=&quot;/Users/immbio/Desktop/Project/Angelina/LNmLToRev/data/LNmLToRev_E18fil_seurat.rds&quot;)</code></pre>
</div>
</div>
<div id="integrate-data-across-location" class="section level2">
<h2>integrate data across location</h2>
<pre class="r"><code>Idents(seuratE18fil) &lt;- seuratE18fil$location

seurat.list &lt;- SplitObject(object = seuratE18fil, split.by = &quot;location&quot;)
for (i in 1:length(x = seurat.list)) {
    seurat.list[[i]] &lt;- NormalizeData(object = seurat.list[[i]],
                                      verbose = FALSE)
    seurat.list[[i]] &lt;- FindVariableFeatures(object = seurat.list[[i]], 
        selection.method = &quot;vst&quot;, nfeatures = 2000, verbose = FALSE)
}

seurat.anchors &lt;- FindIntegrationAnchors(object.list = seurat.list, dims = 1:20)
seuratE18fil.int &lt;- IntegrateData(anchorset = seurat.anchors, dims = 1:20)
DefaultAssay(object = seuratE18fil.int) &lt;- &quot;integrated&quot;

## rerun seurat
seuratE18fil.int &lt;- ScaleData(object = seuratE18fil.int, verbose = FALSE,
                         features = rownames(seuratE18fil.int))
seuratE18fil.int &lt;- RunPCA(object = seuratE18fil.int, npcs = 20, verbose = FALSE)
seuratE18fil.int &lt;- RunTSNE(object = seuratE18fil.int, recuction = &quot;pca&quot;, dims = 1:20)
seuratE18fil.int &lt;- RunUMAP(object = seuratE18fil.int, recuction = &quot;pca&quot;, dims = 1:20)

seuratE18fil.int &lt;- FindNeighbors(object = seuratE18fil.int, reduction = &quot;pca&quot;, dims = 1:20)
res &lt;- c(0.6, 0.8, 0.4, 0.25)
for (i in 1:length(res)){
  seuratE18fil.int &lt;- FindClusters(object = seuratE18fil.int, resolution = res[i],
                              random.seed = 1234)
}</code></pre>
<div id="save-object-fil-int" class="section level3">
<h3>save object fil int</h3>
<pre class="r"><code>saveRDS(seuratE18fil.int, file=&quot;/Users/immbio/Desktop/Project/Angelina/LNmLToRev/data/LNmLToRev_E18fil_integrated_seurat.rds&quot;)</code></pre>
</div>
<div id="load-object-fil-int" class="section level3">
<h3>load object fil int</h3>
<pre class="r"><code>fileNam &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/LNmLToRev/data/LNmLToRev_E18fil_integrated_seurat.rds&quot;
seuratE18fil.int &lt;- readRDS(fileNam)</code></pre>
<pre class="r"><code>DefaultAssay(object = seuratE18fil.int) &lt;- &quot;RNA&quot;
seuratE18fil.int$intCluster &lt;- seuratE18fil.int$integrated_snn_res.0.4
Idents(seuratE18fil.int) &lt;- seuratE18fil.int$intCluster

colPal &lt;- c(&quot;#DAF7A6&quot;, &quot;#FFC300&quot;, &quot;#FF5733&quot;, &quot;#C70039&quot;, &quot;#900C3F&quot;, &quot;#b66e8d&quot;,
            &quot;#61a4ba&quot;, &quot;#6178ba&quot;, &quot;#54a87f&quot;, &quot;#25328a&quot;, &quot;#b6856e&quot;,
            &quot;#ba6161&quot;, &quot;#20714a&quot;, &quot;#0073C2FF&quot;, &quot;#EFC000FF&quot;, &quot;#868686FF&quot;, 
            &quot;#CD534CFF&quot;,&quot;#7AA6DCFF&quot;, &quot;#003C67FF&quot;, &quot;#8F7700FF&quot;, &quot;#3B3B3BFF&quot;,
            &quot;#A73030FF&quot;, &quot;#4A6990FF&quot;)[1:length(unique(seuratE18fil.int$intCluster))]
names(colPal) &lt;- unique(seuratE18fil.int$intCluster)</code></pre>
</div>
</div>
<div id="dimplot-int-data" class="section level2">
<h2>Dimplot int data</h2>
<div id="clustering-2" class="section level3">
<h3>clustering</h3>
<pre class="r"><code>DimPlot(seuratE18fil.int, reduction = &quot;umap&quot;,
        pt.size = 0.1, label = T, shuffle = T, cols = colPal) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;umap1&quot;) +
  ylab(&quot;umap2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN.Rmd/umap%20int%20clustering%200.4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umap-int-clustering-0.4-1">
Past versions of "umap int clustering 0.4-1.png"
</button>
</p>
<div id="fig-no-spaces-umap-int-clustering-0.4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/7eb18ba49db6016e50aa819397bd8c950ca38572/docs/figure/E18_mLN_iLN.Rmd/umap int clustering 0.4-1.png" target="_blank">7eb18ba</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="location-2" class="section level3">
<h3>location</h3>
<pre class="r"><code>DimPlot(seuratE18fil.int, reduction = &quot;umap&quot;, group.by = &quot;location&quot;, cols = collocation,
        pt.size = 0.1, shuffle = T) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;umap1&quot;) +
  ylab(&quot;umap2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN.Rmd/umap%20int%20location-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umap-int-location-1">
Past versions of "umap int location-1.png"
</button>
</p>
<div id="fig-no-spaces-umap-int-location-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/7eb18ba49db6016e50aa819397bd8c950ca38572/docs/figure/E18_mLN_iLN.Rmd/umap int location-1.png" target="_blank">7eb18ba</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="assign-label" class="section level3">
<h3>assign label</h3>
<pre class="r"><code>seuratE18fil.int$label &lt;- &quot;label&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;0&quot;)] &lt;- &quot;Fb1&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;1&quot;)] &lt;- &quot;Prolif&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;2&quot;)] &lt;- &quot;Fb2&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;3&quot;)] &lt;- &quot;Fb3&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;4&quot;)] &lt;- &quot;VSMC&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;5&quot;)] &lt;- &quot;Mesothelial&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;6&quot;)] &lt;- &quot;LN-VSMC&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;7&quot;)] &lt;- &quot;Fb4&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;8&quot;)] &lt;- &quot;Fb5&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;9&quot;)] &lt;- &quot;Neuronal1&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;10&quot;)] &lt;- &quot;Neuroendocrine&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;11&quot;)] &lt;- &quot;Adipocytes&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;12&quot;)] &lt;- &quot;Neuronal2&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;13&quot;)] &lt;- &quot;Neuronal3&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;14&quot;)] &lt;- &quot;Fb6&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;15&quot;)] &lt;- &quot;Epithelial&quot;
seuratE18fil.int$label[which(seuratE18fil.int$intCluster == &quot;16&quot;)] &lt;- &quot;RBC&quot;

table(seuratE18fil.int$label)</code></pre>
<pre><code>
    Adipocytes     Epithelial            Fb1            Fb2            Fb3            Fb4 
          1348             73           9658           4436           3786           2091 
           Fb5            Fb6        LN-VSMC    Mesothelial Neuroendocrine      Neuronal1 
          1695            175           2280           2685           1476           1689 
     Neuronal2      Neuronal3         Prolif            RBC           VSMC 
          1171            447           4790             21           2799 </code></pre>
<pre class="r"><code>##order
seuratE18fil.int$label &lt;- factor(seuratE18fil.int$label, levels = c(&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;, &quot;Fb4&quot;, &quot;Fb5&quot;, &quot;Fb6&quot;, &quot;LN-VSMC&quot;, &quot;VSMC&quot;, &quot;Neuronal1&quot;, &quot;Neuronal2&quot;, &quot;Neuronal3&quot;, &quot;Neuroendocrine&quot;, &quot;Adipocytes&quot;, &quot;Epithelial&quot;, &quot;Mesothelial&quot;, &quot;RBC&quot;, &quot;Prolif&quot;))
table(seuratE18fil.int$label)</code></pre>
<pre><code>
           Fb1            Fb2            Fb3            Fb4            Fb5            Fb6 
          9658           4436           3786           2091           1695            175 
       LN-VSMC           VSMC      Neuronal1      Neuronal2      Neuronal3 Neuroendocrine 
          2280           2799           1689           1171            447           1476 
    Adipocytes     Epithelial    Mesothelial            RBC         Prolif 
          1348             73           2685             21           4790 </code></pre>
<pre class="r"><code>colLab &lt;- c(&quot;#42a071&quot;, &quot;#900C3F&quot;,&quot;#b66e8d&quot;, &quot;#61a4ba&quot;, &quot;#424671&quot;, &quot;#003C67FF&quot;,
            &quot;#e3953d&quot;, &quot;#714542&quot;, &quot;#b6856e&quot;, &quot;#FFC300&quot;, &quot;#8F7700FF&quot;, &quot;#3B3B3BFF&quot;,
            &quot;#A73030FF&quot;, &quot;#4A6990FF&quot;,&quot;#CD534CFF&quot;,&quot;#7AA6DCFF&quot;,&quot;#DAF7A6&quot;)

names(colLab) &lt;- c(unique(seuratE18fil.int$label))</code></pre>
</div>
<div id="label" class="section level3">
<h3>label</h3>
<pre class="r"><code>DimPlot(seuratE18fil.int, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, cols = colLab)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN.Rmd/umap%20int%20label-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umap-int-label-1">
Past versions of "umap int label-1.png"
</button>
</p>
<div id="fig-no-spaces-umap-int-label-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/umap int label-1.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/7eb18ba49db6016e50aa819397bd8c950ca38572/docs/figure/E18_mLN_iLN.Rmd/umap int label-1.png" target="_blank">7eb18ba</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratE18fil.int, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, pt.size=0.5,
        cols = colLab, shuffle = T)+
  theme_void()</code></pre>
<p><img src="figure/E18_mLN_iLN.Rmd/umap%20int%20label-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umap-int-label-2">
Past versions of "umap int label-2.png"
</button>
</p>
<div id="fig-no-spaces-umap-int-label-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/umap int label-2.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/7eb18ba49db6016e50aa819397bd8c950ca38572/docs/figure/E18_mLN_iLN.Rmd/umap int label-2.png" target="_blank">7eb18ba</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seuratE18fil.int, reduction = &quot;umap&quot;, group.by = &quot;label&quot;, pt.size=0.5,
        cols = colLab, shuffle = T)+
  theme_void() + 
  theme(legend.position = &quot;none&quot;) </code></pre>
<p><img src="figure/E18_mLN_iLN.Rmd/umap%20int%20label-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umap-int-label-3">
Past versions of "umap int label-3.png"
</button>
</p>
<div id="fig-no-spaces-umap-int-label-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/7eb18ba49db6016e50aa819397bd8c950ca38572/docs/figure/E18_mLN_iLN.Rmd/umap int label-3.png" target="_blank">7eb18ba</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="calculate-cluster-marker-genes-int" class="section level3">
<h3>calculate cluster marker genes int</h3>
<pre class="r"><code>seuratE18fil.int &lt;- JoinLayers(seuratE18fil.int)
## calculate marker genes
Idents(seuratE18fil.int) &lt;- seuratE18fil.int$intCluster
levels(seuratE18fil.int)</code></pre>
<pre><code> [1] &quot;0&quot;  &quot;1&quot;  &quot;2&quot;  &quot;3&quot;  &quot;4&quot;  &quot;5&quot;  &quot;6&quot;  &quot;7&quot;  &quot;8&quot;  &quot;9&quot;  &quot;10&quot; &quot;11&quot; &quot;12&quot; &quot;13&quot; &quot;14&quot; &quot;15&quot; &quot;16&quot;</code></pre>
<pre class="r"><code>markerGenes_int &lt;- FindAllMarkers(seuratE18fil.int, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01) 

#save table
#write.table(markerGenes_int, 
            #file= &quot;/Users/immbio/Desktop/Project/Angelina/LNmLToRev/data/markerGenes_E18fil_integrated_intCluster&quot;,
            #sep=&quot;\t&quot;,
            #quote=F,
            #row.names=F,
            #col.names=T)</code></pre>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratE18fil.int)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) 

selGenes &lt;- data.frame(geneID=c(&quot;Ptprc&quot;, &quot;Msln&quot;, &quot;Mki67&quot;, &quot;Kcnn3&quot;, &quot;Tcf21&quot;, &quot;Pecam1&quot;, &quot;Lyve1&quot;, &quot;Ccl21a&quot;, &quot;Icam1&quot;, &quot;Cd34&quot;, &quot;Rosa26eyfp&quot;, &quot;Krt19&quot;, &quot;Acta2&quot;,&quot;Myh11&quot;, &quot;Cxcl13&quot;, &quot;Epcam&quot;, &quot;Adipoq&quot;)) %&gt;% 
  left_join(., genes, by = &quot;geneID&quot;) 

pList &lt;- sapply(selGenes$gene, function(x){
  p &lt;- FeaturePlot(seuratE18fil.int, reduction = &quot;umap&quot;, 
            features = x, 
            cols=c(&quot;lightgrey&quot;,&quot;darkred&quot;),
            order = T)+
  theme(legend.position=&quot;right&quot;)
  plot(p)
})</code></pre>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-1">
Past versions of "featureplot sel genes fil int-1.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-1.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-2">
Past versions of "featureplot sel genes fil int-2.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-2"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-2.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-3">
Past versions of "featureplot sel genes fil int-3.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-3"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-3.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-4">
Past versions of "featureplot sel genes fil int-4.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-4"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-4.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-5">
Past versions of "featureplot sel genes fil int-5.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-5"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-5.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-6">
Past versions of "featureplot sel genes fil int-6.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-6"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-6.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-7">
Past versions of "featureplot sel genes fil int-7.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-7"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-7.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-8">
Past versions of "featureplot sel genes fil int-8.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-8"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-8.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-9">
Past versions of "featureplot sel genes fil int-9.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-9"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-9.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-10">
Past versions of "featureplot sel genes fil int-10.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-10"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-10.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-11">
Past versions of "featureplot sel genes fil int-11.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-11"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-11.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-12">
Past versions of "featureplot sel genes fil int-12.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-12"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-12.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-13">
Past versions of "featureplot sel genes fil int-13.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-13"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-13.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-14">
Past versions of "featureplot sel genes fil int-14.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-14"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-14.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-15">
Past versions of "featureplot sel genes fil int-15.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-15"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-15.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-16">
Past versions of "featureplot sel genes fil int-16.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-16"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-16.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/E18_mLN_iLN.Rmd/featureplot%20sel%20genes%20fil%20int-17.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-featureplot-sel-genes-fil-int-17">
Past versions of "featureplot sel genes fil int-17.png"
</button>
</p>
<div id="fig-no-spaces-featureplot-sel-genes-fil-int-17"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/featureplot sel genes fil int-17.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="cluster-characterization" class="section level2">
<h2>cluster characterization</h2>
<div id="heatmap-funcion" class="section level3">
<h3>heatmap funcion</h3>
<pre class="r"><code>avgHeatmap &lt;- function(seurat, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  
  selGenes &lt;- selGenes$gene
  
  ## assay data
  clusterAssigned &lt;- as.data.frame(Idents(seurat)) %&gt;%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seurat)
  
  ## genes of interest
  genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;% filter(geneID %in% selGenes)

  ## matrix with averaged cnts per ident
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;,&quot;&quot;,rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]

  ## color columns according to cluster
  annotation_col &lt;- as.data.frame(gsub(&quot;(^.*?_)&quot;,&quot;&quot;,
                                       colnames(logNormExpresMa)))%&gt;%
    dplyr::mutate(celltype=gsub(&quot;(_.*$)&quot;,&quot;&quot;,colnames(logNormExpresMa)))
  colnames(annotation_col)[1] &lt;- &quot;col1&quot;
  annotation_col &lt;- annotation_col %&gt;%
    dplyr::mutate(cond = gsub(&quot;.*_&quot;,&quot;&quot;,col1)) %&gt;%
    dplyr::select(cond, celltype)
  rownames(annotation_col) &lt;- colnames(logNormExpresMa) 

  ann_colors = list(
      cond = colVecCond,
      celltype=colVecIdent)
  if(is.null(ann_colors$cond)){
    annotation_col$cond &lt;- NULL
  }
  
  ## adjust order
  logNormExpresMa &lt;- logNormExpresMa[selGenes,]
  if(is.null(ordVec)){
    ordVec &lt;- levels(seurat)
  }
  logNormExpresMa &lt;- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale=&quot;row&quot; ,treeheight_row = 0, cluster_rows = cr, 
         cluster_cols = cc,
         color = colorRampPalette(c(&quot;#2166AC&quot;, &quot;#F7F7F7&quot;, &quot;#B2182B&quot;))(50),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}</code></pre>
</div>
<div id="heatmap" class="section level3">
<h3>heatmap</h3>
<pre class="r"><code>### plot DE genes top 10 avg logFC
markerAll &lt;- markerGenes_int %&gt;% group_by(cluster) %&gt;% 
  mutate(geneID = gene) %&gt;% top_n(10, -p_val_adj) %&gt;%
  top_n(10, avg_log2FC) %&gt;%
  mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;,  geneID)) %&gt;% 
  filter(nchar(gene)&gt;1)

grpCnt &lt;- markerAll %&gt;% group_by(cluster) %&gt;% summarise(cnt=n())
gapR &lt;- data.frame(cluster=unique(markerAll$cluster)) %&gt;% 
  left_join(.,grpCnt, by=&quot;cluster&quot;) %&gt;% mutate(cumSum=cumsum(cnt)) 
ordVec &lt;- levels(seuratE18fil.int)

pOut &lt;- avgHeatmap(seurat = seuratE18fil.int, selGenes = markerAll,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecR=gapR$cumSum, gapVecC=NULL,cc=T,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/E18_mLN_iLN.Rmd/characterize%20clusters%20heatmap-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-characterize-clusters-heatmap-1">
Past versions of "characterize clusters heatmap-1.png"
</button>
</p>
<div id="fig-no-spaces-characterize-clusters-heatmap-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/7eb18ba49db6016e50aa819397bd8c950ca38572/docs/figure/E18_mLN_iLN.Rmd/characterize clusters heatmap-1.png" target="_blank">7eb18ba</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dotplot-frc-marker" class="section level3">
<h3>dotplot FRC marker</h3>
<pre class="r"><code>seurat_markers &lt;- data.frame(gene=c(&quot;Fcgr2b&quot;,&quot;Fcer2a&quot;,&quot;Cr2&quot;,&quot;Cxcl13&quot;,
                                    &quot;Slc7a11&quot;, &quot;Ccl19&quot;, 
                                    &quot;Ccl21a&quot;, &quot;Fmod&quot;, &quot;Grem1&quot;, &quot;Bmp4&quot;,
                                    &quot;Tnfsf11&quot;, &quot;Fbn2&quot;,
                                    &quot;Pltp&quot; ,&quot;C1rb&quot;, &quot;Lepr&quot;, &quot;Ptn&quot;,
                                     &quot;Nr4a1&quot;, &quot;Cxcl10&quot;, &quot;Cxcl9&quot;, 
                                    &quot;F3&quot;, &quot;Fbln1&quot;, &quot;Gdf10&quot;, &quot;Adamtsl1&quot;, 
                                    &quot;Col15a1&quot;,  &quot;Cd34&quot;,
                                    &quot;Igfbp6&quot;, &quot;Pi16&quot;, &quot;Thy1&quot;, &quot;Dpp4&quot;, &quot;Sema3c&quot;, 
                                    &quot;Acta2&quot;, &quot;Myh11&quot;, &quot;Mcam&quot;, &quot;Itga7&quot;, &quot;Esam&quot;, &quot;Rgs4&quot;
                                    ))

genes &lt;- data.frame(geneID=rownames(seuratE18fil.int)) %&gt;%
  mutate(gene=gsub(&quot;.*\\.&quot;, &quot;&quot;,  geneID)) 

markerAll &lt;- seurat_markers %&gt;% left_join(., genes, by=&quot;gene&quot;)


## Dotplot all
Idents(seuratE18fil.int) &lt;- seuratE18fil.int$label

DotPlot(seuratE18fil.int, assay=&quot;RNA&quot;, features = rev(markerAll$geneID), scale =T,
        cluster.idents = F) +
  scale_color_viridis_c() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=rev(markerAll$geneID), labels=rev(markerAll$gene)) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;)</code></pre>
<p><img src="figure/E18_mLN_iLN.Rmd/dotplot%20FRC%20marker-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-dotplot-FRC-marker-1">
Past versions of "dotplot FRC marker-1.png"
</button>
</p>
<div id="fig-no-spaces-dotplot-FRC-marker-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/c2baee6ef3a59c90b1c9c550a2398348bd2549b7/docs/figure/E18_mLN_iLN.Rmd/dotplot FRC marker-1.png" target="_blank">c2baee6</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/angeldemartin/LNmLToRev/blob/7eb18ba49db6016e50aa819397bd8c950ca38572/docs/figure/E18_mLN_iLN.Rmd/dotplot FRC marker-1.png" target="_blank">7eb18ba</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Tue May 20 14:36:05 2025&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.7.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
 [4] ggupset_0.4.1               gridExtra_2.3               DOSE_3.30.5                
 [7] enrichplot_1.24.4           msigdbr_10.0.2              org.Hs.eg.db_3.19.1        
[10] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[13] metap_1.12                  scater_1.32.1               scuttle_1.14.0             
[16] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[19] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[22] forcats_1.0.0               stringr_1.5.1               purrr_1.0.4                
[25] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[28] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[31] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.2       
[34] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[37] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.5.0          
[40] pheatmap_1.0.12             ggpubr_0.6.0                ggplot2_3.5.2              
[43] Seurat_5.3.0                SeuratObject_5.1.0          sp_2.2-0                   
[46] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.10.4            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-7            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.4            parallel_4.4.0           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.5                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.4         
 [25] uwot_0.2.3                curl_6.2.2                tidytree_0.4.6           
 [28] mime_0.13                 evaluate_1.0.3            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.16             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.0             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-23              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-3-1         
 [64] reticulate_1.42.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.0           caTools_1.18.3           
 [76] ggtree_3.12.0             data.table_1.17.2         R.oo_1.27.1              
 [79] RSpectra_0.16-2           irlba_2.3.5.1             gridGraphics_0.5-1       
 [82] fastDummies_1.7.5         lazyeval_0.2.2            yaml_2.3.10              
 [85] survival_3.8-3            scattermore_1.2           crayon_1.5.3             
 [88] RcppAnnoy_0.0.22          RColorBrewer_1.1-3        progressr_0.15.1         
 [91] tweenr_2.0.3              later_1.4.2               ggridges_0.5.6           
 [94] codetools_0.2-20          GlobalOptions_0.1.2       aod_1.3.3                
 [97] KEGGREST_1.44.1           Rtsne_0.17                shape_1.4.6.1            
[100] limma_3.60.6              pkgconfig_2.0.3           TMB_1.9.17               
[103] spatstat.univar_3.1-3     mathjaxr_1.8-0            EnvStats_3.1.0           
[106] aplot_0.2.5               scatterplot3d_0.3-44      ape_5.8-1                
[109] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[112] plyr_1.8.9                httr_1.4.7                rbibutils_2.3            
[115] tools_4.4.0               globals_0.18.0            beeswarm_0.4.0           
[118] broom_1.0.8               nlme_3.1-168              lambda.r_1.2.4           
[121] assertthat_0.2.1          lme4_1.1-37               digest_0.6.37            
[124] numDeriv_2016.8-1.1       Matrix_1.7-3              farver_2.1.2             
[127] tzdb_0.5.0                remaCor_0.0.18            reshape2_1.4.4           
[130] yulab.utils_0.2.0         glue_1.8.0                cachem_1.1.0             
[133] polyclip_1.10-7           generics_0.1.4            Biostrings_2.72.1        
[136] mvtnorm_1.3-3             presto_1.0.0              parallelly_1.44.0        
[139] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[142] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[145] pbapply_1.7-2             httr2_1.1.2               spam_2.11-1              
[148] gson_0.1.0                graphlayouts_1.2.2        gtools_3.9.5             
[151] ggsignif_0.6.4            RcppEigen_0.3.4.0.2       shiny_1.10.0             
[154] GenomeInfoDbData_1.2.12   glmmTMB_1.1.11            R.utils_2.13.0           
[157] memoise_2.0.1             rmarkdown_2.29            scales_1.4.0             
[160] R.methodsS3_1.8.2         future_1.49.0             RANN_2.6.2               
[163] Cairo_1.6-2               spatstat.data_3.1-6       rstudioapi_0.17.1        
[166] cluster_2.1.8.1           whisker_0.4.1             mutoss_0.1-13            
[169] spatstat.utils_3.1-3      hms_1.1.3                 fitdistrplus_1.2-2       
[172] cowplot_1.1.3             colorspace_2.1-1          rlang_1.1.6              
[175] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  xts_0.14.1               
[178] dotCall64_1.2             shinydashboard_0.7.3      ggforce_0.4.2            
[181] laeken_0.5.3              mgcv_1.9-3                xfun_0.52                
[184] e1071_1.7-16              TH.data_1.1-3             iterators_1.0.14         
[187] abind_1.4-8               GOSemSim_2.30.2           treeio_1.28.0            
[190] futile.options_1.0.1      bitops_1.0-9              Rdpack_2.6.4             
[193] promises_1.3.2            scatterpie_0.2.4          RSQLite_2.3.11           
[196] qvalue_2.36.0             sandwich_3.1-1            fgsea_1.30.0             
[199] DelayedArray_0.30.1       proxy_0.4-27              GO.db_3.19.1             
[202] compiler_4.4.0            prettyunits_1.2.0         boot_1.3-31              
[205] beachmat_2.20.0           listenv_0.9.1             Rcpp_1.0.14              
[208] edgeR_4.2.2               workflowr_1.7.1           BiocSingular_1.20.0      
[211] tensor_1.5                MASS_7.3-65               progress_1.2.3           
[214] BiocParallel_1.38.0       babelgene_22.9            spatstat.random_3.3-3    
[217] R6_2.6.1                  fastmap_1.2.0             multcomp_1.4-28          
[220] fastmatch_1.1-6           rstatix_0.7.2             vipor_0.4.7              
[223] TTR_0.24.4                ROCR_1.0-11               TFisher_0.2.0            
[226] rsvd_1.0.5                vcd_1.4-13                nnet_7.3-20              
[229] gtable_0.3.6              KernSmooth_2.23-26        miniUI_0.1.2             
[232] deldir_2.0-4              htmltools_0.5.8.1         ggthemes_5.1.0           
[235] bit64_4.6.0-1             spatstat.explore_3.4-2    lifecycle_1.0.4          
[238] blme_1.0-6                msigdbdf_24.1.1           nloptr_2.2.1             
[241] sass_0.4.10               vctrs_0.6.5               robustbase_0.99-4-1      
[244] spatstat.geom_3.3-6       sn_2.1.1                  ggfun_0.1.8              
[247] future.apply_1.11.3       bslib_0.9.0               pillar_1.10.2            
[250] gplots_3.2.0              pcaMethods_1.96.0         locfit_1.5-9.12          
[253] jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.7.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
 [4] ggupset_0.4.1               gridExtra_2.3               DOSE_3.30.5                
 [7] enrichplot_1.24.4           msigdbr_10.0.2              org.Hs.eg.db_3.19.1        
[10] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[13] metap_1.12                  scater_1.32.1               scuttle_1.14.0             
[16] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[19] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[22] forcats_1.0.0               stringr_1.5.1               purrr_1.0.4                
[25] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[28] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[31] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.2       
[34] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[37] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.5.0          
[40] pheatmap_1.0.12             ggpubr_0.6.0                ggplot2_3.5.2              
[43] Seurat_5.3.0                SeuratObject_5.1.0          sp_2.2-0                   
[46] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.10.4            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-7            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.4            parallel_4.4.0           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.5                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.4         
 [25] uwot_0.2.3                curl_6.2.2                tidytree_0.4.6           
 [28] mime_0.13                 evaluate_1.0.3            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.16             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.0             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-23              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-3-1         
 [64] reticulate_1.42.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.0           caTools_1.18.3           
 [76] ggtree_3.12.0             data.table_1.17.2         R.oo_1.27.1              
 [79] RSpectra_0.16-2           irlba_2.3.5.1             gridGraphics_0.5-1       
 [82] fastDummies_1.7.5         lazyeval_0.2.2            yaml_2.3.10              
 [85] survival_3.8-3            scattermore_1.2           crayon_1.5.3             
 [88] RcppAnnoy_0.0.22          RColorBrewer_1.1-3        progressr_0.15.1         
 [91] tweenr_2.0.3              later_1.4.2               ggridges_0.5.6           
 [94] codetools_0.2-20          GlobalOptions_0.1.2       aod_1.3.3                
 [97] KEGGREST_1.44.1           Rtsne_0.17                shape_1.4.6.1            
[100] limma_3.60.6              pkgconfig_2.0.3           TMB_1.9.17               
[103] spatstat.univar_3.1-3     mathjaxr_1.8-0            EnvStats_3.1.0           
[106] aplot_0.2.5               scatterplot3d_0.3-44      ape_5.8-1                
[109] spatstat.sparse_3.1-0     xtable_1.8-4              car_3.1-3                
[112] plyr_1.8.9                httr_1.4.7                rbibutils_2.3            
[115] tools_4.4.0               globals_0.18.0            beeswarm_0.4.0           
[118] broom_1.0.8               nlme_3.1-168              lambda.r_1.2.4           
[121] assertthat_0.2.1          lme4_1.1-37               digest_0.6.37            
[124] numDeriv_2016.8-1.1       Matrix_1.7-3              farver_2.1.2             
[127] tzdb_0.5.0                remaCor_0.0.18            reshape2_1.4.4           
[130] yulab.utils_0.2.0         glue_1.8.0                cachem_1.1.0             
[133] polyclip_1.10-7           generics_0.1.4            Biostrings_2.72.1        
[136] mvtnorm_1.3-3             presto_1.0.0              parallelly_1.44.0        
[139] mnormt_2.1.1              statmod_1.5.0             RcppHNSW_0.6.0           
[142] ScaledMatrix_1.12.0       carData_3.0-5             minqa_1.2.8              
[145] pbapply_1.7-2             httr2_1.1.2               spam_2.11-1              
[148] gson_0.1.0                graphlayouts_1.2.2        gtools_3.9.5             
[151] ggsignif_0.6.4            RcppEigen_0.3.4.0.2       shiny_1.10.0             
[154] GenomeInfoDbData_1.2.12   glmmTMB_1.1.11            R.utils_2.13.0           
[157] memoise_2.0.1             rmarkdown_2.29            scales_1.4.0             
[160] R.methodsS3_1.8.2         future_1.49.0             RANN_2.6.2               
[163] Cairo_1.6-2               spatstat.data_3.1-6       rstudioapi_0.17.1        
[166] cluster_2.1.8.1           whisker_0.4.1             mutoss_0.1-13            
[169] spatstat.utils_3.1-3      hms_1.1.3                 fitdistrplus_1.2-2       
[172] cowplot_1.1.3             colorspace_2.1-1          rlang_1.1.6              
[175] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  xts_0.14.1               
[178] dotCall64_1.2             shinydashboard_0.7.3      ggforce_0.4.2            
[181] laeken_0.5.3              mgcv_1.9-3                xfun_0.52                
[184] e1071_1.7-16              TH.data_1.1-3             iterators_1.0.14         
[187] abind_1.4-8               GOSemSim_2.30.2           treeio_1.28.0            
[190] futile.options_1.0.1      bitops_1.0-9              Rdpack_2.6.4             
[193] promises_1.3.2            scatterpie_0.2.4          RSQLite_2.3.11           
[196] qvalue_2.36.0             sandwich_3.1-1            fgsea_1.30.0             
[199] DelayedArray_0.30.1       proxy_0.4-27              GO.db_3.19.1             
[202] compiler_4.4.0            prettyunits_1.2.0         boot_1.3-31              
[205] beachmat_2.20.0           listenv_0.9.1             Rcpp_1.0.14              
[208] edgeR_4.2.2               workflowr_1.7.1           BiocSingular_1.20.0      
[211] tensor_1.5                MASS_7.3-65               progress_1.2.3           
[214] BiocParallel_1.38.0       babelgene_22.9            spatstat.random_3.3-3    
[217] R6_2.6.1                  fastmap_1.2.0             multcomp_1.4-28          
[220] fastmatch_1.1-6           rstatix_0.7.2             vipor_0.4.7              
[223] TTR_0.24.4                ROCR_1.0-11               TFisher_0.2.0            
[226] rsvd_1.0.5                vcd_1.4-13                nnet_7.3-20              
[229] gtable_0.3.6              KernSmooth_2.23-26        miniUI_0.1.2             
[232] deldir_2.0-4              htmltools_0.5.8.1         ggthemes_5.1.0           
[235] bit64_4.6.0-1             spatstat.explore_3.4-2    lifecycle_1.0.4          
[238] blme_1.0-6                msigdbdf_24.1.1           nloptr_2.2.1             
[241] sass_0.4.10               vctrs_0.6.5               robustbase_0.99-4-1      
[244] spatstat.geom_3.3-6       sn_2.1.1                  ggfun_0.1.8              
[247] future.apply_1.11.3       bslib_0.9.0               pillar_1.10.2            
[250] gplots_3.2.0              pcaMethods_1.96.0         locfit_1.5-9.12          
[253] jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
